src/index.js

```diff
@@ -1,117 +1,164 @@
 import dns from 'dns';
 import setConfig from './config';
-import Context from './context';
-import Callback from './callback';
 import Report from './report';
 import globals from './globals';
 
-function makeDnsPromise(host) {
+const dnsPromises = {};
+
+function getDnsPromise(host) {
+  if (dnsPromises[host]) {
```

Comment:
We re-resolve this vs keeping a cache because if something changes (ex. a host changed between when a lambda was last invoked), we want it re-resolved